compute_tools/src/bin/fast_import.rs

```diff
@@ -83,6 +87,189 @@ const DEFAULT_LOCALE: &str = if cfg!(target_os = "macos") {
+            PostgresMajorVersion::V16 => 16,
+            PostgresMajorVersion::V17 => 17,
+        };
+        postgres_initdb::do_run_initdb(postgres_initdb::RunInitdbArgs {
+            superuser: initdb_user,
+            locale: DEFAULT_LOCALE, // XXX: this shouldn't be hard-coded,
+            pg_version,
+            initdb_bin: self.pg_bin_dir.join("initdb").as_ref(),
+            library_search_path: &self.pg_lib_dir, // TODO: is this right? Prob works in compute image, not sure about neon_local.
+            pgdata: &self.pgdata_dir,
+        })
+        .await
+        .context("initdb")
+    }
+
+    async fn start(
+        &mut self,
+        initdb_user: &str,
+        port: u16,
+        nproc: usize,
+    ) -> Result<&tokio::process::Child, anyhow::Error> {
+        self.prepare(initdb_user).await?;
+
+        //
+        // Launch postgres process
+        //
+        let mut proc = tokio::process::Command::new(&self.pgbin)
+            .arg("-D")
+            .arg(&self.pgdata_dir)
+            .args(["-p", &format!("{port}")])
+            .args(["-c", "wal_level=minimal"])
+            .args(["-c", "shared_buffers=10GB"])
+            .args(["-c", "max_wal_senders=0"])
+            .args(["-c", "fsync=off"])
+            .args(["-c", "full_page_writes=off"])
+            .args(["-c", "synchronous_commit=off"])
+            .args(["-c", "maintenance_work_mem=8388608"])
+            .args(["-c", &format!("max_parallel_maintenance_workers={nproc}")])
+            .args(["-c", &format!("max_parallel_workers={nproc}")])
+            .args(["-c", &format!("max_parallel_workers_per_gather={nproc}")])
+            .args(["-c", &format!("max_worker_processes={nproc}")])
+            .args(["-c", "effective_io_concurrency=100"])
+            .env_clear()
+            .env("LD_LIBRARY_PATH", &self.pg_lib_dir)
+            .stdout(std::process::Stdio::piped())
+            .stderr(std::process::Stdio::piped())
+            .spawn()
+            .context("spawn postgres")?;
+
+        info!("spawned postgres, waiting for it to become ready");
+        tokio::spawn(
+            child_stdio_to_log::relay_process_output(proc.stdout.take(), proc.stderr.take())
+                .instrument(info_span!("postgres")),
+        );
+
+        self.postgres_proc = Some(proc);
+        Ok(self.postgres_proc.as_ref().unwrap())
+    }
+
+    async fn shutdown(&mut self) -> Result<(), anyhow::Error> {
+        let proc: &mut tokio::process::Child = self.postgres_proc.as_mut().unwrap();
+        info!("shutdown postgres");
+        {
+            nix::sys::signal::kill(
+                Pid::from_raw(i32::try_from(proc.id().unwrap()).expect("convert child pid to i32")),
+                nix::sys::signal::SIGTERM,
+            )
+            .context("signal postgres to shut down")?;
+            proc.wait()
+                .await
+                .context("wait for postgres to shut down")?;
+        }
+        Ok(())
```

Comment:
Why the extra scope? You could instead do. Also, is the return code not interesting?
```
  199     async fn shutdown(&mut self) -> Result<(), anyhow::Error> {
  200         let proc: &mut tokio::process::Child = self.postgres_proc.as_mut().unwrap();
  201         info!("shutdown postgres");
  202         nix::sys::signal::kill(
  203             Pid::from_raw(i32::try_from(proc.id().unwrap()).expect("convert child pid to i32")),
  204             nix::sys::signal::SIGTERM,
  205         )
  206         .context("signal postgres to shut down")?;
  207         proc.wait()
  208             .await
  209             .context("wait for postgres to shut down")
  210             .map(|_| ())
  211     }
  ```